controllers/seedimage_controller.go

```diff
@@ -200,9 +196,7 @@ func (r *SeedImageReconciler) reconcileBuildImagePod(ctx context.Context, seedIm
 
 		// pod is already there and with the right configuration
 		if foundPod.Annotations["elemental.cattle.io/base-image"] == podBaseImg &&
-			foundPod.Annotations["elemental.cattle.io/registration-url"] == podRegURL &&
-			foundPod.Annotations["elemental.cattle.io/cloud-config-b64"] == podCloudConfigb64 {
-
+			foundPod.Annotations["elemental.cattle.io/configmap-uid"] == string(podConfigMap.ObjectMeta.UID) {
```

Comment:
we don't need this: we track in the annotations the config in order to check if the conf is changed on the seedimage. If so, we recreate the build pod, otherwise what we have is already up-to-date.

We create the configmap only once, so this condition will be always true.

Since now we track the MachineRegistration as the owner, we should probably check that the MachineRegistration reference in the SeedImage is still the same as the owner (otherwise, update the owner and rebuild the Pod). Anyway, this could be for another PR.

So, here, I would just drop the configmap-uid annotation and keep only the base-image one.